### PR TITLE
Feat(eos_cli_config_gen): Add support for eos_cli under router_ospf process ids

### LINKED ISSUE #3009: Feat(eos_cli_config_gen): Add support for OSPF Process `eos_cli` Escape Hatch
### Enhancement summary

Similar to BGP defaults, add a key for `ospf_defaults` whose value will be a list of eos cli statements to be configured under an OSPF process ID. 

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

This is relevant when a user wants to add some OSPF configuration under a process ID used for underlay routing which isn't covered by the existing **eos_cli_config_gen** `router_ospf` data model.

### Describe the solution you would like

A data model similar to BGP defaults

```
router_ospf:
  process_ids:
    - id: <int>
      ospf_defaults:
        - <eos-cli-command-1>
        - <eos-cli-command-2>
        ...
```

### Describe alternatives you have considered

_No response_

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by KyleMui**
## Change Summary

Add eos cli to router ospf process ids

## Related Issue(s)

Fixes #3009  

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
Added the below to the eos_cli_config_gen `router_ospf` data model
```
router_ospf:
  process_ids:
    - id: <int>
      eos_cli: |
        <eos-cli-command-1>
        <eos-cli-command-2>
```

## How to test

Added to eos_cli_config_gen molecule scenario `router-ospf` host-var:
```
router_ospf:
  process_ids:
    - id: 101
       eos_cli: |
         area 5 not-so-stubby lsa type-7 convert type-5
         network 3.2.4.1/8 area 3.2.4.1
```

renders to:
```
router ospf 101
   area 5 not-so-stubby lsa type-7 convert type-5
   network 3.2.4.1/8 area 3.2.4.1
```

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

**Comment by jrecchia1029** [comment]
So it looks like the type of `eos_cli` should be a string just like the other `eos_cli` keys in the **eos_cli_config_gen** role.  The description for the key should match those descriptions too.

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/eos_cli_config_gen.schema.yml [hunk 1]
```diff
@@ -10004,6 +10004,11 @@ keys:
                           max: 16777215
             mpls_ldp_sync_default:
               type: bool
+            eos_cli:
+              type: list
+              items:
+                type: str
+                description: The eos cli command
```
Comment (jrecchia1029) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/eos_cli_config_gen.schema.yml]: Make this description for this the same as the other `eos_cli` keys.

i.e. 

```
Multiline string with EOS CLI rendered under the ospf process id
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_ospf.schema.yml [hunk 2]
```diff
@@ -298,7 +298,5 @@ keys:
             mpls_ldp_sync_default:
               type: bool
             eos_cli:
-              type: list
-              items:
-                type: str
-                description: The eos cli command
+              type: str
+              description: Multiline EOS CLI rendered directly on the router ospf in the final EOS configuration
```
Comment (jrecchia1029) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_ospf.schema.yml]: It's under a process id which is under router ospf. So I would just clarify in comment `Multiline EOS CLI rendered directly on the router ospf process id in the final EOS configuration`

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_ospf.schema.yml [hunk 3]
```diff
@@ -297,3 +297,6 @@ keys:
                           max: 16777215
             mpls_ldp_sync_default:
               type: bool
+            eos_cli:
+              type: str
+              description: Multiline EOS CLI rendered directly on the router ospf process id in the final EOS configuration
```
Comment (gmuloc) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/router_ospf.schema.yml]:
```suggestion
              description: Multiline EOS CLI rendered directly on the Router OSPF process ID in the final EOS configuration
```

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/router-ospf.yml [hunk 4]
```diff
@@ -57,6 +57,9 @@ router_ospf:
           initial: 100
           min: 200
           max: 300
+      eos_cli:
+        area 5 not-so-stubby lsa type-7 convert type-5
+        network 3.2.4.1 0.255.255.255 area 3.2.4.1
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/router-ospf.yml]: This is what will appear in the config on the device (EOS 4.27)
```suggestion
        network 3.0.0.0/8 area 3.2.4.1
```

Though, yes the configuration supplied will be accepted by the device.

**Review by gmuloc — CHANGES_REQUESTED** [review]
One little change in the description in the schema.

While this PR is a correct addition and it _can_ make sense to add an escape hatch `eos_cli` in this place, this **should** not replace implementing proper model for missing OSPF commands in `eos_cli_config_gen`.

In this PR the following lines are being used for the test:
```
   area 5 not-so-stubby lsa type-7 convert type-5
   network 3.2.4.1 0.255.255.255 area 3.2.4.1
```

Please note that `network` is already supported using `network_prefixes` which is anyway what will appear in the rendered configuration. What is the reason to use the CLI that will be converted instead?

```
Spine1(config-router-ospf)#network 3.2.4.1 0.255.255.255 area 3.2.4.1
Spine1(config-router-ospf)#!
Spine1(config-router-ospf)#show active
router ospf 666
   network 3.0.0.0/8 area 3.2.4.1
```

It seems what is really missing today in `eos_cli_config_gen` is the `not-so-stuby` type of areas - is this a correct statement?

Could you please open an issue with your intended configuration to add the missing key properly to `eos_cli_config_gen` as well?


> The issue being referenced has been closed by @ClausHolbechArista with the same kind of comments as mine: https://github.com/aristanetworks/ansible-avd/issues/3009#issuecomment-1607705472
He is suggesting adding `eos_cli` only as a fix for missing model.

**Review by jrecchia1029 — COMMENTED** [review]
LGTM

**Comment by jrecchia1029** [comment]
> One little change in the description in the schema.
> 
> While this PR is a correct addition and it _can_ make sense to add an escape hatch `eos_cli` in this place, this **should** not replace implementing proper model for missing OSPF commands in `eos_cli_config_gen`.
> 
> In this PR the following lines are being used for the test:
> 
> ```
>    area 5 not-so-stubby lsa type-7 convert type-5
>    network 3.2.4.1 0.255.255.255 area 3.2.4.1
> ```
> 
> Please note that `network` is already supported using `network_prefixes` which is anyway what will appear in the rendered configuration. What is the reason to use the CLI that will be converted instead?
> 
> ```
> Spine1(config-router-ospf)#network 3.2.4.1 0.255.255.255 area 3.2.4.1
> Spine1(config-router-ospf)#!
> Spine1(config-router-ospf)#show active
> router ospf 666
>    network 3.0.0.0/8 area 3.2.4.1
> ```
> 
> It seems what is really missing today in `eos_cli_config_gen` is the `not-so-stuby` type of areas - is this a correct statement?
> 
> Could you please open an issue with your intended configuration to add the missing key properly to `eos_cli_config_gen` as well?
> 
> > The issue being referenced has been closed by @ClausHolbechArista with the same kind of comments as mine: [#3009 (comment)](https://github.com/aristanetworks/ansible-avd/issues/3009#issuecomment-1607705472)
> > He is suggesting adding `eos_cli` only as a fix for missing model.

The commands used in the test were chosen arbitrarily.  This PR is less about a specific set of commands not covered by `eos_cli_config_gen` and more about offering that escape hatch you mentioned.  I believe the issue which Claus closed already mentions being open to using the `eos_cli` key instead of `ospf_defaults`

**Review by jrecchia1029 — APPROVED** [review]

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/router-ospf.yml [hunk 5]
```diff
@@ -57,6 +57,9 @@ router_ospf:
           initial: 100
           min: 200
           max: 300
+      eos_cli: |
+        area 5 not-so-stubby lsa type-7 convert type-5
+        network 3.2.4.1/8 area 3.2.4.1
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/router-ospf.yml]: let's change this example as it is not really a good one

**Comment by ClausHolbechArista** [comment]
Please update the PR description to reflect the data model being implemented.

**Review by gmuloc — APPROVED** [review]

**Comment by gmuloc** [comment]
Thanks @KyleMui for this contribution

**Pull request merged**